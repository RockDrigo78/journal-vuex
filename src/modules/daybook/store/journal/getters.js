export const getEntriesByTerm =
    (state) =>
    (terms = '') => {
        if (terms.length === 0) return state.entries;

        return state.entries.filter((entry) =>
            entry.text.toLowerCase().includes(terms.toLowerCase())
        );
    };
export const getEntryById = (state) => (id) => {
    return state.entries.find((entry) => entry.id === id);
};
