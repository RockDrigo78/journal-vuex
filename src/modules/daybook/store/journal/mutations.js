export const setEntries = (state, entries) => {
    state.entries = [...state.entries, ...entries];
    state.isLoading = false;
};

export const updateEntry = (state, entry) => {
    const idIndex = state.entries.map((e) => e.id).indexOf(entry.id);
    state.entries[idIndex] = entry;
};
export const addEntry = (state, entry) => {
    state.entries = [entry, ...state.entries];
    state.isLoading = false;
};

export const removeEntry = (state, id) => {
    state.entries = state.entries.filter((entry) => entry.id !== id);
};
