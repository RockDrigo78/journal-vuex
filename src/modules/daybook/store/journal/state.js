export default () => ({
    isLoading: true,
    entries: [
        {
            id: new Date().getTime(),
            date: new Date().toDateString(),
            text: 'Eu ullamco aute magna do dolore id consectetur magna. Ad consectetur nulla non dolore cupidatat eiusmod exercitation laborum fugiat. Officia dolor mollit qui sunt exercitation minim duis tempor excepteur.',
            picture: null,
        },
        {
            id: new Date().getTime() + 100,
            date: new Date().toDateString(),
            text: 'Non enim amet sint dolor incididunt. Eu et velit nisi nisi irure et ex ad sunt ullamco et. Velit duis reprehenderit nisi ullamco. Elit fugiat minim eu eiusmod do reprehenderit fugiat est nulla cillum eu commodo. Aute deserunt in eu ipsum sint sint ea eu deserunt irure. Ea quis aliqua cillum qui.',
            picture: null,
        },
        {
            id: new Date().getTime() + 200,
            date: new Date().toDateString(),
            text: 'Adipisicing adipisicing quis cillum anim occaecat ipsum excepteur anim reprehenderit esse. Non duis proident non deserunt et labore velit ex sunt magna proident laborum. Sit ipsum consectetur pariatur est consequat consequat anim aute.',
            picture: null,
        },
        {
            id: new Date().getTime() + 300,
            date: new Date().toDateString(),
            text: 'Veniam nostrud eu ea voluptate officia ex consectetur ullamco ex ipsum. Laborum culpa commodo proident magna ea eu laboris eiusmod aliqua commodo. Mollit et Lorem non occaecat occaecat ex. Eu dolore tempor labore minim est laboris anim aute. Ullamco deserunt consequat ullamco cillum qui ea consequat et laborum officia sint. Ad ipsum irure enim nulla consectetur dolore esse cillum irure aliquip aute non. Minim ipsum et minim dolore pariatur.',
            picture: null,
        },
    ],
});
