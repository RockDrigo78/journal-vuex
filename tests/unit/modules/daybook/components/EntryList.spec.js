import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';

import journal from '@/modules/daybook/store/journal';
import EntryList from '@/modules/daybook/components/EntryList.vue';
import { journalState } from '../../../mock-data/test-journal-state';

const createVuexStore = (initialState) =>
    createStore({
        modules: {
            journal: {
                ...journal,
                state: { ...initialState },
            },
        },
    });

describe('EntryList component tests', () => {
    const store = createVuexStore(journalState);

    const mockRouter = {
        push: jest.fn(),
    };

    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallowMount(EntryList, {
            global: {
                mocks: {
                    $router: mockRouter,
                },
                plugins: [store],
            },
        });
    });

    test('should call getEntriesByTerm and show 2 entries', () => {
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.findAll('entry-stub').length).toBe(2);
    });

    test('should call getEntriesByTerm and filter entries', async () => {
        const input = wrapper.find('input');
        await input.setValue('data 1');

        expect(wrapper.findAll('entry-stub').length).toBe(1);
    });

    test('new button should redirect to /new', () => {
        wrapper.find('button').trigger('click');

        expect(mockRouter.push).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: 'new',
            },
        });
    });
});
