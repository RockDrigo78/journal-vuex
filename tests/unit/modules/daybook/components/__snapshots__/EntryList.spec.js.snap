// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryList component tests should call getEntriesByTerm and show 2 entries 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Search Entry"></div>
  <div>
    <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> New Entry </button></div>
  </div>
  <div class="entry-scrollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
